stdlib/public/core/StringBridge.swift

```diff
@@ -737,6 +737,22 @@ internal func _SwiftCreateBridgedString_DoNotCall(
   return Unmanaged<AnyObject>.passRetained(str._bridgeToObjectiveCImpl())
 }
 
+extension String {
+  @_spi(Foundation)
+  public init<Encoding: Unicode.Encoding>(_immortalCocoaString: AnyObject, buffer: UnsafeBufferPointer<UInt8>, encoding: Encoding.Type) {
+    precondition(encoding == Unicode.ASCII.self || encoding == Unicode.UTF8.self)
+    let storage = __SharedStringStorage(
+      immortalCocoa ptr: buffer.baseAddress!,
+      owner: _immortalCocoaString,
```

Comment:
It would be neat to figure out a way to assert that this is actually immortal on the ObjC side